File: Corte 1/reto4.py
"""
RETO 4
Simulacion del juego "Piedra, Papel o Tijera" contra la maquina.
Reglas:
- Piedra vence a Tijera
- Tijera vence a Papel
- Papel vence a Piedra
"""

import random


def main():
    print("   RETO 4 - Piedra, Papel o Tijera")
    print("Opciones: 1) Piedra   2) Papel   3) Tijera")

    opcion_jugador = int(input("Elige tu jugada (1, 2 o 3): "))

    # Validamos que la opcion sea correcta
    while opcion_jugador < 1 or opcion_jugador > 3:
        opcion_jugador = int(input("Opcion invalida. Elige 1, 2 o 3: "))

    # La maquina genera su jugada aleatoriamente (1, 2 o 3)
    opcion_maquina = random.randint(1, 3)

    nombres = {1: "Piedra", 2: "Papel", 3: "Tijera"}

    print("-------------------------------------")
    print(f"Tu elegiste: {nombres[opcion_jugador]}")
    print(f"La maquina eligio: {nombres[opcion_maquina]}")
    print("-------------------------------------")

    # Determinamos el resultado
    if opcion_jugador == opcion_maquina:
        print("Resultado: EMPATE")
    elif (
        (opcion_jugador == 1 and opcion_maquina == 3) or  # Piedra vence a Tijera
        (opcion_jugador == 3 and opcion_maquina == 2) or  # Tijera vence a Papel
        (opcion_jugador == 2 and opcion_maquina == 1)      # Papel vence a Piedra
    ):
        print("Resultado: GANASTE!")
    else:
        print("Resultado: PERDISTE.")

    print("=====================================")


if __name__ == "__main__":
    main()

File: Corte 1/Retos/reto4.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
#include <map>
#include <string>
using namespace std;

/*
 * RETO 4
 * Simulacion del juego "Piedra, Papel o Tijera" contra la maquina.
 * Reglas:
 * - Piedra vence a Tijera
 * - Tijera vence a Papel
 * - Papel vence a Piedra
 */

int main() {
    // Inicializamos la semilla de numeros aleatorios con la hora actual
    srand(static_cast<unsigned int>(time(0)));

    int opcionJugador;

    cout << "   RETO 4 - Piedra, Papel o Tijera" << endl;
    cout << "Opciones: 1) Piedra   2) Papel   3) Tijera" << endl;
    cout << "Elige tu jugada (1, 2 o 3): ";
    cin >> opcionJugador;

    // Validamos que la opcion sea correcta
    while (opcionJugador < 1 || opcionJugador > 3) {
        cout << "Opcion invalida. Elige 1, 2 o 3: ";
        cin >> opcionJugador;
    }

    // La maquina genera su jugada aleatoriamente (1, 2 o 3)
    int opcionMaquina = (rand() % 3) + 1;

    map<int, string> nombres = {{1, "Piedra"}, {2, "Papel"}, {3, "Tijera"}};

    cout << "-------------------------------------" << endl;
    cout << "Tu elegiste: " << nombres[opcionJugador] << endl;
    cout << "La maquina eligio: " << nombres[opcionMaquina] << endl;
    cout << "-------------------------------------" << endl;

    // Determinamos el resultado
    if (opcionJugador == opcionMaquina) {
        cout << "Resultado: EMPATE" << endl;
    } else if (
        (opcionJugador == 1 && opcionMaquina == 3) ||  // Piedra vence a Tijera
        (opcionJugador == 3 && opcionMaquina == 2) ||  // Tijera vence a Papel
        (opcionJugador == 2 && opcionMaquina == 1)     // Papel vence a Piedra
    ) {
        cout << "Resultado: GANASTE!" << endl;
    } else {
        cout << "Resultado: PERDISTE." << endl;
    }

    cout << "=====================================" << endl;

    return 0;
}